File: tests/c/TestTemplate.c
#define main pattern_main
#include "../../oop_in_c/Template/Template.c"
#undef main
#include <stdio.h>

int main() {
  printf("Testing C Template...\n");
  // Call pattern main
  pattern_main();
  printf("C Template Test Passed!\n");
  return 0;
}

File: tests/java/TestTemplate.java
public class TestTemplate {
  public static void main(String[] args) {
    System.out.println("Testing Java Template...");
    try {
      Client.main(args);
    } catch (Exception e) {
      e.printStackTrace();
      System.exit(1);
    }
    System.out.println("Java Template Test Passed!");
  }
}
